src/care/DefaultMacros.h

```diff
@@ -60,7 +60,7 @@
 /// @arg[in] CHECK The variable to check that the start and end macros match
 ///
 ////////////////////////////////////////////////////////////////////////////////
-#define CARE_CHECKED_FOR_LOOP_START(INDEX, START_INDEX, END_INDEX, CHECK) for (int INDEX = START_INDEX; INDEX < END_INDEX; ++INDEX) CARE_NEST_BEGIN(CHECK)
+#define CARE_CHECKED_FOR_LOOP_START(INDEX, START_INDEX, END_INDEX, CHECK) for (int INDEX = START_INDEX; INDEX < (int)END_INDEX; ++INDEX) CARE_NEST_BEGIN(CHECK)
```

Comment:
In ale3d, the users have added some loops where END_INDEX is a size_t ( vector.size() ). That is what motivated this cast to int

Reply:
Without this cast, is there a compiler warning/error?

Reply:
And does this only show up in the legacy build?

Reply:
Yes, in care compatibility mode it errored out, only in legacy. I was thinking of perhaps changing this to something fancy rather than a cast (I don't know if something like decltype could determine if we need a size_t or an int?), but I have not tried that yet.

Right now I'm changing things in my ale3d pull request to get the LEGACY flag passed to care properly, so please don't approve or merge this in until I have that resolved.

Reply:
Can you undo this change for now?

Reply:
Nevermind, we assume int anyway.